crates/solver/src/settlement_rater.rs

```diff
@@ -225,16 +228,16 @@ impl SettlementRating for SettlementRater {
         }
 
         let earned_fees = settlement.total_earned_fees(prices);
+        let gas_estimate = settlement.gas_amount.unwrap_or(simulation.gas_estimate);
```

Comment:
Isn't this dangerous in the case of pre/post hooks the additional gas will not be taken into consideration and therefore the settlement may fail when submitted with the solver provided gas amount?

Reply:
So, AFAIS there are two options here:
1. Send pre and post interactions to solvers and expect from solvers to include them into gas estimation.
2. Simulate settlement with stripped pre and post interactions in the driver and the difference in gas usage with and without interactions is the `gas_estimate` for interactions (more or less). 

I am not sure if pre/post interactions can be simulated in isolation meaningfully so that we know the gas usage beforehand, in that case we could send the `gas_for_hooks` to solvers or just append the value to their gas_estimate in driver, without even sending anything to solvers.

Reply:
(1) might be problematic for solvers that expect driver to generate call_data (aka Quasilabs if I am not wrong)

Reply:
We wanted to include it in the order `cost` field (so we communicate to the solver the additional gas units tied to the order).

Reply:
When I think about it more, I feel like instead of bothering solvers with estimating gas (unless they are co-located), we should just fix our gas estimates. Given the cool new wrapper contracts we have for computing exact gas amounts, can we not use that instead of `eth_estimateGas` to get a better understanding of the gas a solution will use and not require solvers to pass us this information?